### PR TITLE
Refactor/user data structure in local storage

### LINKED ISSUE #50: change how data is stored in localStorage to array of users.
Right now only one user is stored in local storage and if logout happens, user is wiped from local storage.

Goals of this ticket:
-refine data such that an array of users is saved in local storage until storage is wiped.
-change login method to verify if user is stored in local storage, and the password entered is correct.

### LINKED ISSUE #52: Make register method check if user already in local storage

### PR DESCRIPTION
**Pull request by arositen**
## What type of PR is this? (check all applicable)

- [X] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Refactor of how UserContext handles and checks user data. Main changes:

- localStorage now can store a list of users. i.e. a "fake database"
- user and useList is consistent on refresh
- register method checks for a unique username before approving new users and adding to userList
- login method validates username and password match before login
- logout method doesn't remove currentUser, just sets to null
- updateBalance method updates userList, so balance can be found even after user logs out and logs back in.
-  id generated by uuid
- user objects updated to match outline provided by Yusuke:
   - interface User {
        id: string;
        username: string;
        password: string;
        balance: number;
        order: Order[];
   }

Minor changes:
- got rid of testing console logs
- changed height styling of login and register modals to make it easier to test
- added uuid library

## Related Tickets & Documents

- Closes #50 
- Closes #52 

## QA Instructions, Screenshots, Recordings

Just try to use all functions. Register a user, add some money, try logging out then logging in, refresh, etc.

### CODE REVIEW

**Review by MatthewNeie — APPROVED** [review]

File: src/components/molecules/registerModal.jsx [hunk 1]
```diff
@@ -11,15 +11,13 @@ function RegisterModal({ isOpen, handleClose }) {
...
-    console.log(username);
-    console.log(password);
     register(username, password);
     handleClose();
   };
 
   return (
     isOpen && (
-      <div className="absolute top-0 left-0 flex justify-center items-center h-full w-full bg-black bg-opacity-25 z-30">
+      <div className="absolute top-0 left-0 flex justify-center items-center h-[100vh] w-full bg-black bg-opacity-25 z-30">
```
Comment (yusuken1121) [inline src/components/molecules/registerModal.jsx:20]: You don’t necessarily have to change it, but I thought changing absolute to fixed would provide a better user experience by keeping the modal fixed in place relative to scrolling

File: src/components/molecules/loginModal.jsx [hunk 2]
```diff
@@ -11,14 +11,12 @@ function LoginModal({ isOpen, handleClose }) {
...
         e.preventDefault();
-        console.log(username);
-        console.log(password);
         login(username, password);
         handleClose();
     }
 
     return (
-        (isOpen && <div className='absolute top-0 left-0 flex justify-center items-center h-full w-full bg-black bg-opacity-25 z-30'>
+        (isOpen && <div className='absolute top-0 left-0 flex justify-center items-center h-[100vh] w-full bg-black bg-opacity-25 z-30'>
```
Comment (yusuken1121) [inline src/components/molecules/loginModal.jsx:19]: You don’t necessarily have to change it, but I thought changing absolute to fixed would provide a better user experience by keeping the modal fixed in place relative to scrolling

File: src/contexts/UserContext.jsx [hunk 3]
```diff
@@ -8,41 +9,77 @@ export function useUser() {
 
 export const UserProvider = ({ children }) => {
 
+    const [userList, setUserList] = useState([]);
     const [user, setUser] = useState(null);
 
+    //initialize user list in localStorage
+    if (!localStorage.getItem('userList')) {
+        localStorage.setItem('userList', JSON.stringify(userList));
+    };
```
Comment (yusuken1121) [inline src/contexts/UserContext.jsx:18]: I don’t think you need to include the semicolon.

**Comment by yusuken1121** [comment]
I think this is very well-organized and excellent code!
Thank you.
I made a few minor comments, but I will approve it.
I would appreciate it if you could consider them in this or the next PR if you think they are needed.

**Review by yusuken1121 — APPROVED** [review]

File: src/components/molecules/registerModal.jsx (hunk 1 above)
Comment (arositen) [reply src/components/molecules/registerModal.jsx:20]: Yes totally agree, that's already in my next branch!

**Pull request merged**